roles/manage_cnp/tasks/add_function.yml

```diff
@@ -0,0 +1,26 @@
+---
+- name: Validate cnpg_namespace
+  ansible.builtin.fail:
+    fail: msg="The variable 'cnpg_namespace' is not defined and empty"
+  when: (cngp_namespace is defined) and (cngp_namespace | length < 1)
+
+- name: Validate pod_name
+  ansible.builtin.fail:
+    fail: msg="The variable 'pod_name' is not defined and empty"
+  when: (pod_name is defined) and (pod_name | length < 1)
+
+- name: Validate db_name
+  ansible.builtin.fail:
+    fail: msg="The variable 'db_name' is not defined and empty"
+  when: (db_name is defined) and (db_name | length < 1)
+
+- name: Validate add_function_sql
+  ansible.builtin.fail:
+    fail: msg="The variable 'add_function_sql' is not defined and empty"
+  when: (add_function_sql is defined) and (add_function_sql | length < 1)
+
+- name: Add a Function
+  kubernetes.core.k8s_exec:
+    namespace: "{{ cnpg_namespace }}"
+    pod: "{{ pod_name }}"
+    command: psql -d "{{ db_name }}" -c "{{ add_function_sql }}"
```

Comment:
With the `kubernetes.core` collection, the k8's inventory and/or kubectl connection files are necessary to run a command. This is included in the CNP setup roles under `/files/cnpg_sandbox_monitored_cluster.yml`. This role needs to be tested to ensure all the kubernetes collection requirements are met and connection is successful.

Reply:
tested